include/modules/sv_detection_methods/analyze_split_read_method.hpp

```diff
@@ -36,6 +36,32 @@ void split_string(std::string const & str, Container & cont, char const delim =
  */
 void retrieve_aligned_segments(std::string const & sa_string, std::vector<AlignedSegment> & aligned_segments);
 
+/*! \brief Calculates the end respectively start position of two consecutive mates (current and next
+ *         [AlignedSegment](\ref AlignedSegment)) depending on their orientation and corrects the position of the second
+ *         mate, if they are overlapping.
+ *
+ * \param[in] current          - current [AlignedSegment](\ref AlignedSegment)
+ * \param[in] next             - consecutive next [AlignedSegment](\ref AlignedSegment)
+ * \param[in] distance_on_read - distance between the consecutive aligned_segments on the read
+ *
+ * \returns std::tie(mate1_pos, mate2_pos) - a tuple of the resulting mate positions
+ *
+ * \details \verbatim
+    Case 1: --current--> --next-->
+    Case 2: --current--> <--next--
+    Case 3: <--current-- --next-->
+    Case 4: <--current-- <--next--
+                       | |
+               mate1_pos mate2_pos
+
+    overlapping Case: --current-->         distance_on_read < 0
+                                --next-->
+                                 ||
+                         mate1_pos mate2_pos
+    \endverbatim
+ */
+std::tuple<int32_t, int32_t> get_mate_positions(AlignedSegment current, AlignedSegment next, int32_t distance_on_read);
```

Comment:
```suggestion
std::pair<int32_t, int32_t> get_mate_positions(AlignedSegment current, AlignedSegment next, int32_t distance_on_read);
```

Reply:
Just as a note to think about: Actually, this function doesn't need to be declared in the header file, if it is only accessed locally from the same cpp file. Then, of course, it cannot be included/used from other files or tests.